spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/diagnostics/analyzer/NoSuchBeanDefinitionFailureAnalyzerTests.java

```diff
@@ -166,8 +166,8 @@ public void failureAnalysisForNoMatchName() {
 		assertThat(analysis.getDescription()).startsWith(String.format(
 				"Constructor in %s required a bean named '%s' that could not be found",
 				StringNameHandler.class.getName(), "test-string"));
-		assertThat(analysis.getDescription().contains(
-				"No matching auto-configuration has been found for this bean name."));
+		assertThat(analysis.getDescription()).contains(
+				"No matching auto-configuration has been found for this bean name.");
```

Comment:
This assertion fails.

@snicoll Do you remember what the intention behind this test was? The current description only matches the assertion above.

Reply:
I am not following. The assertion makes sure that no matching auto-config was found. Why do you think it can be removed?

Reply:
@snicoll The actual value of `analysis.getDescription()` is `"Constructor in org.springframework.boot.autoconfigure.diagnostics.analyzer.NoSuchBeanDefinitionFailureAnalyzerTests$StringNameHandler required a bean named 'test-string' that could not be found."

That is why the assertion fails. This didn't attract attention because `assertThat` wrapped the whole expression instead of just `analysis.getDescription()` and thus effectively didn't assert anything.

Now I was asking if the [assertion in line 166](https://github.com/pvorb/spring-boot/blob/15b5d5e6b1d1be76fee13227356d24e5cef531c6/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/diagnostics/analyzer/NoSuchBeanDefinitionFailureAnalyzerTests.java#L166-L168) isn't enough to verify the description and if the line I changed can be removed.